api/v1alpha1/proxydefaults_types_test.go

```diff
@@ -86,11 +91,25 @@ func TestProxyDefaults_MatchesConsul(t *testing.T) {
 					},
 				},
 			},
+			Matches: true,
+		},
+		"mismatched types does not match": {
+			Ours: ProxyDefaults{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: common.Global,
+				},
+				Spec: ProxyDefaultsSpec{},
+			},
+			Theirs: &capi.ServiceConfigEntry{
+				Name: common.Global,
+				Kind: capi.ProxyDefaults,
+			},
+			Matches: false,
 		},
 	}
 	for name, c := range cases {
 		t.Run(name, func(t *testing.T) {
-			require.True(t, c.Ours.MatchesConsul(c.Theirs))
+			require.Equal(t, c.Ours.MatchesConsul(c.Theirs), c.Matches)
```

Comment:
```suggestion
			require.Equal(t,  c.Matches, c.Ours.MatchesConsul(c.Theirs))
```
exp should be before act